test-data/unit/check-partially-defined.test

```diff
@@ -269,10 +269,14 @@ def f0() -> None:
         if first_iter:
             first_iter = False
             x = 0
-        else:
+        elif int():
             # This is technically a false positive but mypy isn't smart enough for this yet.
             y = x  # E: Name "x" may be undefined
+        else:
+            y = x  # E: Name "x" may be undefined
```

Comment:
Actually maybe add another test case for this? When you have branches like these inside `else` of another `if` that also defines `x` (without the loop). Or did it work already before this PR? (But even if yes, and we didn't have such test I think ti makes sense to add it).

Reply:
I've added some extra test cases that I think should cover what you are talking about. Let me know if not!

Reply:
Looks good! Thanks!